infra/base-images/base-builder/precompile_centipede

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash -eu
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+echo -n "Precompiling centipede"
+
+# Build Centipede with bazel.
+cd "$SRC/centipede/"
+./install_dependencies_debian.sh
+bazel build -c opt :all
+
+apt uninstall curl gnupg apt-transport-https git bazel binutils libssl-dev
+
+# Prepare the weak symbols trick
+cp "$CENTIPEDE_SRC/bazel-bin/libweak_sancov_stubs.so" "$OUT/weak.o"
```

Comment:
Sorry why do we need this in OSS-Fuzz? If we use clang-15 do we still need it? I can't understand why centipede is broken enough that we need this, this should be done upstream, not here.

Reply:
If I understand it correctly, this is not because of `Centipede`, but because we build the target binary and the intermediate binaries with the same cflags.
Recall that we need to build an intermediate binary `lemon` and use it to build the target `sqlite3`. Flag `-fsanitize=trace-loads` is necessary when building `sqlite3`, but unnecessary for `lemon` as it is merely for auxiliary purposes and does not need instrumentation.
However, we build them with the same flags (i.e., `CLFAGS` and `CXXFLAGS`), which makes `lemon` requires the definition of `__sanitizer_cov_load4` (and similar ones) as well. The weak symbols exist so that `lemon` won't throw errors like this:
```/src/sqlite3/bld/../tool/lemon.c:523: undefined reference to `__sanitizer_cov_load4'```